### PR TITLE
Add sql trigger templates

### PR DESCRIPTION
**Pull request by MaddyDev**
Added Sql Trigger templates:

image

image

### CODE REVIEW

File: Build/package-lock.json [hunk 1]
```diff
@@ -2212,6 +2212,12 @@
       "resolved": "https://registry.npmjs.org/arr-union/-/arr-union-3.1.0.tgz",
       "integrity": "sha1-45sJrqne+Gao8gbiiK9jkZuuOcQ="
     },
+    "array-differ": {
```
Comment (Charles-Gagnon) [inline Build/package-lock.json:2215]: Did you mean to update this? The changes look unrelated.

File: Functions.Templates/Documentation/sqlTrigger.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#### Settings for an Azure SQL trigger binding
```
Comment (Charles-Gagnon) [inline Functions.Templates/Documentation/sqlTrigger.md:1]: Is there a reason you made this H4? We should generally start with H1 and go out from there

File: Functions.Templates/Documentation/sqlTrigger.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            [SqlTrigger("[dbo].[Employees]", "SqlConnectionString")]
+            IReadOnlyList<SqlChange<Employees>> changes,
+            ILogger logger)
+        {
+            // The output is used to inspect the trigger binding parameter in test methods.
+            logger.LogInformation("SQL Changes: " + JsonConvert.SerializeObject(changes));
+        }
+```
+
+#### SQL trigger JavaScript example
```
Comment (Charles-Gagnon) [inline Functions.Templates/Documentation/sqlTrigger.md:28]: Is there a reason you did just C# and JS?

File: Functions.Templates/Templates/SqlTrigger-CSharp-Isolated/.template.config/template.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
+{
+    "author": "Microsoft",
+    "classifications": [
+        "Azure Function",
+        "Trigger",
+        "SQL"
+    ],
+    "name": "SqlTriggerBindingIsolated",
+    "identity": "Azure.Function.CSharp.Isolated.SqlTriggerBinding.2.x",
```
Comment (Charles-Gagnon) [inline Functions.Templates/Templates/SqlTrigger-CSharp-Isolated/.template.config/template.json:9]: This should be 3.x, right? (I assume this is supposed to match the version)

```suggestion
    "identity": "Azure.Function.CSharp.Isolated.SqlTriggerBinding.3.x",
```

File: Functions.Templates/Templates/SqlTrigger-CSharp/.template.config/template.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
+{
+    "author": "Microsoft",
+    "classifications": [
+        "Azure Function",
+        "Trigger",
+        "SQL"
+    ],
+    "name": "SqlTriggerBinding",
+    "identity": "Azure.Function.CSharp.SqlTriggerBinding.2.x",
```
Comment (Charles-Gagnon) [inline Functions.Templates/Templates/SqlTrigger-CSharp/.template.config/template.json:9]:
```suggestion
    "identity": "Azure.Function.CSharp.SqlTriggerBinding.3.x",
```

File: Functions.Templates/Templates/SqlTrigger-CSharp-Isolated/SqlTriggerBindingCSharp.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    public static class SqlTriggerBindingCSharp
+    {
+        private readonly ILogger _logger;
+
+        public SqlTriggerBindingCSharp(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<SqlTriggerBindingCSharp>();
+        }
+
+        [Function("SqlTriggerBindingCSharp")]
```
Comment (Charles-Gagnon) [inline Functions.Templates/Templates/SqlTrigger-CSharp-Isolated/SqlTriggerBindingCSharp.cs:18]:
```suggestion
        // Visit https://aka.ms/sqltrigger to learn how to use this trigger binding
        [Function("SqlTriggerBindingCSharp")]
```

File: Functions.Templates/Templates/SqlTrigger-CSharp/function.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
+  "bindings": [
+    {
+      "authLevel": "function",
+      "name": "changes",
+      "type": "sqlTrigger",
+      "direction": "in",
+      "methods": [
```
Comment (Charles-Gagnon) [inline Functions.Templates/Templates/SqlTrigger-CSharp/function.json:8]: Aren't methods and authLevel only for http triggers?

File: Functions.Templates/Templates/SqlTrigger-CSharp/function.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
+{
+  "bindings": [
+    {
+      "authLevel": "function",
+      "name": "changes",
+      "type": "sqlTrigger",
+      "direction": "in",
```
Comment (Charles-Gagnon) [inline Functions.Templates/Templates/SqlTrigger-CSharp/function.json:7]: This seems to be missing the sql trigger values (tableName, connectionStringSetting)

File: Functions.Templates/Templates/SqlTrigger-JavaScript/index.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Sample SQL Trigger Binding
+ * See https://aka.ms/sqltrigger for more information about using this binding
+ *
+ * These tasks should be completed prior to running :
+ *      1. Update "tableName" in function.json - this should be the table that is monitored for changes and triggers/invokes the function.
+ *      2. Add an app setting named "SqlConnectionString" containing the connection string
+ *          to use for the SQL connection
+ *      3. Change the bundle name in host.json to "Microsoft.Azure.Functions.ExtensionBundle.Preview" and the version to "[4.*, 5.0.0)"
```
Comment (Charles-Gagnon) [inline Functions.Templates/Templates/SqlTrigger-JavaScript/index.js:9]: Make sure to add a post-GA task for updating this

File: Functions.Templates/Templates/SqlTrigger-JavaScript/index.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+ * @param context The Azure Function runtime context
+ * @param req The HttpRequest that triggered this function
+ * @param results The array of objects returned by the SQL input binding
```
Comment (Charles-Gagnon) [inline Functions.Templates/Templates/SqlTrigger-JavaScript/index.js:12]: These params aren't correct

File: Functions.Templates/Templates/SqlTrigger-PowerShell/run.ps1 [hunk 11]
```diff
@@ -0,0 +1,22 @@
+<#
+Sample SQL Input Binding
```
Comment (Charles-Gagnon) [inline Functions.Templates/Templates/SqlTrigger-PowerShell/run.ps1:2]: Wrong description

File: Build/package-lock.json (hunk 1 above)
Comment (MaddyDev) [reply Build/package-lock.json:2215]: I did not update this manually, probably the build command did while I was testing.

File: Functions.Templates/Templates/SqlTrigger-CSharp-Isolated/.template.config/template.json (hunk 4 above)
Comment (MaddyDev) [reply Functions.Templates/Templates/SqlTrigger-CSharp-Isolated/.template.config/template.json:9]: This version thing probably isn't even needed. @soninaren Can you please confirm and I can update accoridingly. Thanks.

File: Functions.Templates/Templates/SqlTrigger-JavaScript/index.js (hunk 9 above)
Comment (MaddyDev) [reply Functions.Templates/Templates/SqlTrigger-JavaScript/index.js:9]: Created https://github.com/Azure/azure-functions-sql-extension/issues/872 to track that, thanks.

**Review by lucyzhang929 — APPROVED** [review]

File: Build/package-lock.json (hunk 1 above)
Comment (Charles-Gagnon) [reply Build/package-lock.json:2215]: @soninaren I assume this should be reverted? It seems like something that should be updated in another PR if needed

**Review by Charles-Gagnon — APPROVED** [review]

File: Functions.Templates/Templates/SqlTrigger-CSharp-Isolated/.template.config/template.json (hunk 4 above)
Comment (MaddyDev) [reply Functions.Templates/Templates/SqlTrigger-CSharp-Isolated/.template.config/template.json:9]: @fabiocav Can you please confirm if the version is needed.

Comment (fabiocav) [reply Functions.Templates/Templates/SqlTrigger-CSharp-Isolated/.template.config/template.json:9]: Not needed. You just need to make sure this is a unique identifier.
